flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/IncrementalSourceReader.java

```diff
@@ -141,6 +142,11 @@ public void notifyCheckpointComplete(long checkpointId) throws Exception {
     protected void onSplitFinished(Map<String, SourceSplitState> finishedSplitIds) {
         for (SourceSplitState splitState : finishedSplitIds.values()) {
             SourceSplitBase sourceSplit = splitState.toSourceSplit();
+            if (sourceConfig.getStartupOptions().startupMode == StartupMode.SNAPSHOT_ONLY
+                    && sourceSplit.isStreamSplit()) {
+                // when startupMode = SNAPSHOT_ONLY. the stream split could finish.
+                continue;
+            }
```

Comment:
```suggestion
            if (sourceConfig.getStartupOptions().isSnapshotOnly() && sourceSplit.isStreamSplit()) {
                // when startupMode = SNAPSHOT_ONLY. the stream split could finish.
                continue;
            }
```

Shall we remove the `streamSplit` from `uncompletedStreamSplits` here ?

Reply:
I think it is no need. `uncompletedStreamSplits` puts streamSplit which not start and wait for lacked table metadata. When all metadata are received, will add back in `requestStreamSplitMetaIfNeeded`. So a finished split must not in `uncompletedStreamSplits`.
You mean like `suspendedBinlogSplit` in mysql? We don't support it until add new scan table ability in cdc framework.